postgresql/resource_postgresql_grant_role_test.go

```diff
@@ -0,0 +1,175 @@
+
+	cases := []struct {
+		resource   *schema.ResourceData
+		privileges []string
+		expected   string
+	}{
+		{
+			resource: schema.TestResourceDataRaw(t, resourcePostgreSQLGrantRole().Schema, map[string]interface{}{
+				"role":       roleName,
+				"grant_role": grantRoleName,
+			}),
+			expected: fmt.Sprintf("GRANT %s TO %s", pq.QuoteIdentifier(grantRoleName), pq.QuoteIdentifier(roleName)),
+		},
+		{
+			resource: schema.TestResourceDataRaw(t, resourcePostgreSQLGrantRole().Schema, map[string]interface{}{
+				"role":              roleName,
+				"grant_role":        grantRoleName,
+				"with_admin_option": false,
+			}),
+			expected: fmt.Sprintf("GRANT %s TO %s", pq.QuoteIdentifier(grantRoleName), pq.QuoteIdentifier(roleName)),
+		},
+		{
+			resource: schema.TestResourceDataRaw(t, resourcePostgreSQLGrantRole().Schema, map[string]interface{}{
+				"role":              roleName,
+				"grant_role":        grantRoleName,
+				"with_admin_option": true,
+			}),
+			expected: fmt.Sprintf("GRANT %s TO %s WITH ADMIN OPTION", pq.QuoteIdentifier(grantRoleName), pq.QuoteIdentifier(roleName)),
+		},
+	}
+
+	for _, c := range cases {
+		out := createGrantRoleQuery(c.resource)
+		if out != c.expected {
+			t.Fatalf("Error matching output and expected: %#v vs %#v", out, c.expected)
+		}
+	}
+}
+
+func TestRevokeRoleQuery(t *testing.T) {
+	var roleName = "foo"
+	var grantRoleName = "bar"
+
+	cases := []struct {
+		resource   *schema.ResourceData
+		privileges []string
+		expected   string
+	}{
+		{
+			resource: schema.TestResourceDataRaw(t, resourcePostgreSQLGrantRole().Schema, map[string]interface{}{
+				"role":       roleName,
+				"grant_role": grantRoleName,
+			}),
+			expected: fmt.Sprintf("REVOKE %s FROM %s", pq.QuoteIdentifier(grantRoleName), pq.QuoteIdentifier(roleName)),
+		},
+		{
+			resource: schema.TestResourceDataRaw(t, resourcePostgreSQLGrantRole().Schema, map[string]interface{}{
+				"role":              roleName,
+				"grant_role":        grantRoleName,
+				"with_admin_option": false,
+			}),
+			expected: fmt.Sprintf("REVOKE %s FROM %s", pq.QuoteIdentifier(grantRoleName), pq.QuoteIdentifier(roleName)),
+		},
+		{
+			resource: schema.TestResourceDataRaw(t, resourcePostgreSQLGrantRole().Schema, map[string]interface{}{
+				"role":              roleName,
+				"grant_role":        grantRoleName,
+				"with_admin_option": false,
+			}),
+			expected: fmt.Sprintf("REVOKE %s FROM %s", pq.QuoteIdentifier(grantRoleName), pq.QuoteIdentifier(roleName)),
+		},
+	}
+
+	for _, c := range cases {
+		out := createRevokeRoleQuery(c.resource)
+		if out != c.expected {
+			t.Fatalf("Error matching output and expected: %#v vs %#v", out, c.expected)
+		}
+	}
+}
+
+func TestAccPostgresqlGrantRole(t *testing.T) {
+	skipIfNotAcc(t)
+
+	config := getTestConfig(t)
+	dsn := config.connStr("postgres")
+
+	dbSuffix, teardown := setupTestDatabase(t, false, true)
+	defer teardown()
+
+	_, roleName := getTestDBNames(dbSuffix)
+
+	grantedRoleName := "foo"
```

Comment:
This one could also be created with a `postgresql_role` in `testAccPostgresqlGrantRoleResources` directly so you don't have to create/destroy it yourself.